### PR TITLE
Delete From support for Empty Table

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
Addressed bug found in https://github.com/dolthub/dolt/issues/1923

### CODE REVIEW

**Comment by andrew-wm-arthur** [comment]
Something doesn't add up here, why is an `emptyTable` being returned?

**Comment by VinaiRachakonda** [comment]
I should've left this in WIP because I wanted to think it through again in the morning but the problem is coming from [here](https://github.com/dolthub/go-mysql-server/blob/4b21301f5a15ee849e580f1f6d45d3e78eec9eb9/sql/analyzer/optimization_rules.go#L305): 

The test case essentially simulates an bad filter (one that has an impossible condition) on a delete query. This was the problem with this [issue](https://github.com/dolthub/dolt/issues/1923).

File: sql/plan/delete.go [hunk 1]
```diff
@@ -95,6 +95,11 @@ func (p *DeleteFrom) Database() string {
 
 // RowIter implements the Node interface.
 func (p *DeleteFrom) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	// If an empty table is passed in (potentially from a bad filter) return an empty row iter.
```
Comment (andrew-wm-arthur) [inline sql/plan/delete.go:98]: A cleaner solution would be to have emptyTable implement the deletable interface so we don't need to special case it here

**Review by andrew-wm-arthur — APPROVED** [review]

**Review by andrew-wm-arthur — COMMENTED** [review]
see comment above

**Pull request merged**